scripts/masterCron.ts

```diff
@@ -58,6 +63,110 @@ const getTopOfPeriod = async (provider: any, period: number) => {
   return topOfPeriod;
 };
 
+async function getStrikePrice(
+  vault: Contract,
+  vaultLifecycle: Contract,
+  strikeSelection: Contract,
+  ierc20ABI: any
+) {
+  let expiry;
+  let currentOption = (await vault.optionState()).currentOption;
+  if (currentOption == address(0)) {
+    expiry = await vaultLifecycle.getNextFriday(
+      (
+        await provider.getBlock("latest")
+      ).timestamp
+    );
+  } else {
+    expiry = await vaultLifecycle.getNextFriday(
+      await new ethers.Contract(
+        currentOption,
+        ierc20ABI,
+        provider
+      ).expiryTimestamp()
+    );
+  }
+
+  let isPut = (await vault.vaultParams()).isPut;
+
+  let strike = await strikeSelection.getStrikePrice(expiry, isPut);
+
+  return expiry, isPut, strike;
+}
+
+async function getOptionPremium(
```

Comment:
copying from logic in vault lifecycle [commitAndClose](https://github.com/ribbon-finance/ribbon-v2/blob/bd9b69f2b68132cea8d418709c00acab57e815e0/contracts/libraries/VaultLifecycle.sol#L75)